### PR TITLE
SAK-29445 - Make launch text on LTI tool more user friendly

### PR DESCRIPTION
**Pull request by mcarropellicer**
SAK-29445 - Make launch text on LTI tool more user friendly

### CODE REVIEW

**Comment by ern** [comment]
Miguel,

This contribution was from Francisco Segura can you please update the commit.

Thanks

File: basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java [hunk 1]
```diff
@@ -238,9 +242,12 @@ public void doView(RenderRequest request, RenderResponse response)
 						text.append(windowOpen+"\n");
 						text.append("</script>\n");
 					}
-					text.append(rb.getString("new.page.launch"));
-					text.append("<br><a href=\""+iframeUrl+"\" onclick=\""+windowOpen+"\" target=\"BasicLTI\">"+rb.getString("noiframe.press.here")+"</a>");
+					String siteName = ServerConfigurationService.getString(SITE_NAME, SAKAI);
+					title = title!=null ? title : rb.getString("tool.name", "your tool");
```
Comment (buckett) [inline basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java]: I think title needs to be escaped as it's put into a HTML attribute and should escape any quotes.

Comment (mcarropellicer) [inline basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java]: Should i make the change or should i wait for a new patch? 

Thanks Matthew.

Comment (buckett) [inline basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java]: Are you ok to make the improvement?

Comment (mcarropellicer) [inline basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java]: Done!

File: basiclti/basiclti-portlet/src/bundle/basiclti.properties [hunk 2]
```diff
@@ -3,7 +3,8 @@ noiframes = Your browser does not support iframes.
 noiframe.press.here = Press here to launch.
 not.configured = This tool has not yet been configured.
 
-new.page.launch = Content opened in a new window. Make sure pop up windows are allowed in your browser.
+new.page.launch = (We attempted to launch {0} in a new window. If you wish this to happen automatically in future you need to configure your browser to allow {1} to open pop-up windows.)
+tool.name = your tool
```
Comment (ottenhoff) [inline basiclti/basiclti-portlet/src/bundle/basiclti.properties]: "in the future" instead of "in future"

Comment (mcarropellicer) [inline basiclti/basiclti-portlet/src/bundle/basiclti.properties]: Changed.

File: basiclti/basiclti-portlet/src/bundle/basiclti.properties [hunk 3]
```diff
@@ -3,7 +3,8 @@ noiframes = Your browser does not support iframes.
 noiframe.press.here = Press here to launch.
 not.configured = This tool has not yet been configured.
 
-new.page.launch = Content opened in a new window. Make sure pop up windows are allowed in your browser.
+new.page.launch = (We attempted to launch {0} in a new window. If you wish this to happen automatically in the future you need to configure your browser to allow {1} to open pop-up windows.)
```
Comment (ottenhoff) [inline basiclti/basiclti-portlet/src/bundle/basiclti.properties]: To launch pop-up windows automatically in the future, configure your browser to allow {1} to open pop-up windows.

Comment (mcarropellicer) [inline basiclti/basiclti-portlet/src/bundle/basiclti.properties]: Suggestion included, thanks Sam.

**Pull request merged**